tests/unit/preprocessor/_weighting/test_weighting_landsea_fraction.py

```diff
@@ -0,0 +1,231 @@
+    (CUBE_3, EMPTY_FX_FILES, 'sea', False, CUBE_3),
+    (CUBE_3, L_FX_FILES, 'land', True, CUBE_3_L),
+    (CUBE_3, L_FX_FILES, 'land', False, CUBE_3_L),
+    (CUBE_3, L_FX_FILES, 'sea', True, CUBE_3_O),
+    (CUBE_3, L_FX_FILES, 'sea', False, CUBE_3_O),
+    (CUBE_3, O_FX_FILES, 'land', True, ValueError),
+    (CUBE_3, O_FX_FILES, 'land', False, CUBE_3),
+    (CUBE_3, O_FX_FILES, 'sea', True, ValueError),
+    (CUBE_3, O_FX_FILES, 'sea', False, CUBE_3),
+    (CUBE_3, FX_FILES, 'land', True, CUBE_3_L),
+    (CUBE_3, FX_FILES, 'land', False, CUBE_3_L),
+    (CUBE_3, FX_FILES, 'sea', True, CUBE_3_O),
+    (CUBE_3, FX_FILES, 'sea', False, CUBE_3_O),
+    (CUBE_3, FX_FILES, 'wrong', True, TypeError),
+    (CUBE_3, FX_FILES, 'wrong', False, TypeError),
+    (CUBE_4, {}, 'land', True, ValueError),
+    (CUBE_4, {}, 'land', False, CUBE_4),
+    (CUBE_4, {}, 'sea', True, ValueError),
+    (CUBE_4, {}, 'sea', False, CUBE_4),
+    (CUBE_4, EMPTY_FX_FILES, 'land', True, ValueError),
+    (CUBE_4, EMPTY_FX_FILES, 'land', False, CUBE_4),
+    (CUBE_4, EMPTY_FX_FILES, 'sea', True, ValueError),
+    (CUBE_4, EMPTY_FX_FILES, 'sea', False, CUBE_4),
+    (CUBE_4, L_FX_FILES, 'land', True, ValueError),
+    (CUBE_4, L_FX_FILES, 'land', False, CUBE_4),
+    (CUBE_4, L_FX_FILES, 'sea', True, ValueError),
+    (CUBE_4, L_FX_FILES, 'sea', False, CUBE_4),
+    (CUBE_4, O_FX_FILES, 'land', True, CUBE_4_L),
+    (CUBE_4, O_FX_FILES, 'land', False, CUBE_4_L),
+    (CUBE_4, O_FX_FILES, 'sea', True, CUBE_4_O),
+    (CUBE_4, O_FX_FILES, 'sea', False, CUBE_4_O),
+    (CUBE_4, FX_FILES, 'land', True, CUBE_4_L),
+    (CUBE_4, FX_FILES, 'land', False, CUBE_4_L),
+    (CUBE_4, FX_FILES, 'sea', True, CUBE_4_O),
+    (CUBE_4, FX_FILES, 'sea', False, CUBE_4_O),
+    (CUBE_4, FX_FILES, 'wrong', True, TypeError),
+    (CUBE_4, FX_FILES, 'wrong', False, TypeError),
+]
+
+
+@pytest.mark.parametrize('cube,fx_files,area_type,strict,out',
+                         WEIGHTING_LANDSEA_FRACTION)
+@mock.patch.object(weighting, 'iris', autospec=True)
+def test_weighting_landsea_fraction(mock_iris,
+                                    cube,
+                                    fx_files,
+                                    area_type,
+                                    strict,
+                                    out):
+    """Test landsea fraction weighting preprocessor."""
+    # Exceptions
+    if isinstance(out, type):
+        with pytest.raises(out):
+            weighted_cube = weighting.weighting_landsea_fraction(
+                cube, fx_files, area_type, strict=strict)
+        return
+
+    # Regular cases
+    fx_cubes = []
+    if fx_files.get('sftlf'):
+        fx_cubes.append(CUBE_SFTLF)
+    if fx_files.get('sftof'):
+        fx_cubes.append(CUBE_SFTOF)
+    mock_iris.load_cube.side_effect = fx_cubes
+    weighted_cube = weighting.weighting_landsea_fraction(
+        cube, fx_files, area_type, strict=strict)
+    assert weighted_cube == cube
+    assert weighted_cube is cube
```

Comment:
ooh this is dangerous, this will most probably fail since if you do `c1 = iris.load_cube(file)` and `c2 = iris.load_cube(file)` then `assert c1 == c2` will pass whereas `assert c1 is c2` will fail

Reply:
No, the weighting happens in-place, so I want that `cube_out is cube_in` (no new object is created). Also, the tests are not failing, right? :wink:

Reply:
yes, was a bit confused as to why the tests were passing, have missed the fact that the cube is modified in place :grin: